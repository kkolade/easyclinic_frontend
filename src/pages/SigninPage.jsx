import { useDocumentTitle } from '@mantine/hooks';

import AppShell from 'components/AppShell';
import SigninForm from 'components/SigninForm';

const SigninPage = () => {
  useDocumentTitle('Sign in - EasyClinic');

  return (
    <AppShell>
      <SigninForm />
    </AppShell>
  );
};

export default SigninPage;
