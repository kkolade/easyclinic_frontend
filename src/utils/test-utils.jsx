/* eslint-disable import/no-extraneous-dependencies */

import { MantineProvider } from '@mantine/core';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';
import store from '../redux/store';
import theme from '../theme';

const renderWithProviders = ({ children }) => (
  <Provider store={store}>
    <MantineProvider withNormalizeCSS withGlobalStyles theme={theme}>
      <Router>{children}</Router>
    </MantineProvider>
  </Provider>
);

const customRender = (ui, options) => render(ui, { wrapper: renderWithProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
