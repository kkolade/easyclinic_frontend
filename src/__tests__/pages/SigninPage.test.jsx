import { render } from 'test-utils';

import SigninPage from 'pages/SigninPage';

describe('SigninPage', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<SigninPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
