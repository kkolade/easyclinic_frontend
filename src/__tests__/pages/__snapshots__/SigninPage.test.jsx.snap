// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SigninPage should render correctly 1`] = `
<DocumentFragment>
  <div
    class="mantine-AppShell-root mantine-1udmbix"
  >
    <div
      class="mantine-hr6aor mantine-AppShell-body"
    >
      <nav
        class="mantine-Navbar-root mantine-13ge6fs"
        data-hidden="true"
      >
        <div
          class="mantine-qyxvtq"
        >
          <div
            class="mantine-Image-root mantine-lgo8bz"
            style="width: 12.5rem;"
          >
            <figure
              class="mantine-qenwvq mantine-Image-figure"
            >
              <div
                class="mantine-1iugybl mantine-Image-imageWrapper"
              >
                <img
                  alt="EasyClinic Logo"
                  class="mantine-1i6l9bv mantine-Image-image"
                  src="/logo.png"
                  style="object-fit: cover; width: 12.5rem; height: auto;"
                />
              </div>
            </figure>
          </div>
        </div>
        <div
          class="mantine-hzudtq"
        >
          <div>
            <a
              class="mantine-UnstyledButton-root mantine-NavLink-root mantine-165mgsf"
              data-active="true"
              href="/"
            >
              <span
                class="mantine-1l4mhad mantine-NavLink-icon"
              >
                <svg
                  class="tabler-icon tabler-icon-home"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 12l-2 0l9 -9l9 9l-2 0"
                  />
                  <path
                    d="M5 12v7a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-7"
                  />
                  <path
                    d="M9 21v-6a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v6"
                  />
                </svg>
              </span>
              <span
                class="mantine-dstkpx mantine-NavLink-body"
              >
                <span
                  class="mantine-Text-root mantine-NavLink-label mantine-1sdv04i"
                >
                  Home
                </span>
                <span
                  class="mantine-Text-root mantine-NavLink-description mantine-h4oguy"
                  data-active="true"
                />
              </span>
            </a>
            <div
              aria-hidden="true"
              class="mantine-1avyp1d"
              style="box-sizing: border-box; display: none; height: 0px; overflow: hidden;"
            >
              <div
                style="opacity: 0; transition: opacity 200ms ease;"
              >
                <div
                  class="mantine-1c04k7q mantine-NavLink-children"
                />
              </div>
            </div>
            <a
              class="mantine-UnstyledButton-root mantine-NavLink-root mantine-165mgsf"
              href="/signin"
            >
              <span
                class="mantine-1l4mhad mantine-NavLink-icon"
              >
                <svg
                  class="tabler-icon tabler-icon-login"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M14 8v-2a2 2 0 0 0 -2 -2h-7a2 2 0 0 0 -2 2v12a2 2 0 0 0 2 2h7a2 2 0 0 0 2 -2v-2"
                  />
                  <path
                    d="M20 12h-13l3 -3m0 6l-3 -3"
                  />
                </svg>
              </span>
              <span
                class="mantine-dstkpx mantine-NavLink-body"
              >
                <span
                  class="mantine-Text-root mantine-NavLink-label mantine-1sdv04i"
                >
                  Sign in
                </span>
                <span
                  class="mantine-Text-root mantine-NavLink-description mantine-h4oguy"
                />
              </span>
            </a>
            <div
              aria-hidden="true"
              class="mantine-1avyp1d"
              style="box-sizing: border-box; display: none; height: 0px; overflow: hidden;"
            >
              <div
                style="opacity: 0; transition: opacity 200ms ease;"
              >
                <div
                  class="mantine-1c04k7q mantine-NavLink-children"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="mantine-khtkeg"
        >
          <div
            class="mantine-Text-root mantine-1p5v0xp"
          >
            Â© 2023 EasyClinic
          </div>
        </div>
      </nav>
      <main
        class="mantine-bzbzc1 mantine-AppShell-main"
      >
        <div
          class="mantine-16jzalw"
        >
          <form
            class="mantine-gr49qu"
          >
            <div
              class="mantine-1x8qhrp"
            >
              <h3
                class="mantine-Text-root mantine-Title-root mantine-1fla5xb"
              >
                Sign in
              </h3>
              <div>
                <div
                  class="mantine-InputWrapper-root mantine-TextInput-root mantine-12pjkll"
                >
                  <label
                    class="mantine-InputWrapper-label mantine-TextInput-label mantine-1fzet7j"
                    for="mantine-lfyzfg5n9"
                    id="mantine-lfyzfg5n9-label"
                  >
                    Username
                  </label>
                  <div
                    class="mantine-Input-wrapper mantine-TextInput-wrapper mantine-1v7s5f8"
                  >
                    <input
                      aria-invalid="false"
                      class="mantine-Input-input mantine-TextInput-input mantine-1153slh"
                      id="mantine-lfyzfg5n9"
                      placeholder="Your username"
                      required=""
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="mantine-InputWrapper-root mantine-PasswordInput-root mantine-12pjkll"
                >
                  <label
                    class="mantine-InputWrapper-label mantine-PasswordInput-label mantine-1fzet7j"
                    for="mantine-tz9kvyc0k"
                    id="mantine-tz9kvyc0k-label"
                  >
                    Password
                  </label>
                  <div
                    class="mantine-Input-wrapper mantine-PasswordInput-wrapper mantine-1v7s5f8"
                  >
                    <div
                      aria-invalid="false"
                      class="mantine-PasswordInput-input mantine-Input-input mantine-PasswordInput-input mantine-yqkfp1"
                    >
                      <input
                        class="mantine-fmcwfl mantine-PasswordInput-innerInput"
                        id="mantine-tz9kvyc0k"
                        placeholder="Your password"
                        required=""
                        type="password"
                        value=""
                      />
                    </div>
                    <div
                      class="mantine-c7k433 mantine-Input-rightSection mantine-PasswordInput-rightSection"
                    >
                      <button
                        aria-hidden="true"
                        class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-PasswordInput-visibilityToggle mantine-ki20zf"
                        tabindex="-1"
                        type="button"
                      >
                        <svg
                          fill="none"
                          style="width: 0.9375rem; height: 0.9375rem;"
                          viewBox="0 0 15 15"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M7.5 11C4.80285 11 2.52952 9.62184 1.09622 7.50001C2.52952 5.37816 4.80285 4 7.5 4C10.1971 4 12.4705 5.37816 13.9038 7.50001C12.4705 9.62183 10.1971 11 7.5 11ZM7.5 3C4.30786 3 1.65639 4.70638 0.0760002 7.23501C-0.0253338 7.39715 -0.0253334 7.60288 0.0760014 7.76501C1.65639 10.2936 4.30786 12 7.5 12C10.6921 12 13.3436 10.2936 14.924 7.76501C15.0253 7.60288 15.0253 7.39715 14.924 7.23501C13.3436 4.70638 10.6921 3 7.5 3ZM7.5 9.5C8.60457 9.5 9.5 8.60457 9.5 7.5C9.5 6.39543 8.60457 5.5 7.5 5.5C6.39543 5.5 5.5 6.39543 5.5 7.5C5.5 8.60457 6.39543 9.5 7.5 9.5Z"
                            fill="currentColor"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <button
                class="mantine-UnstyledButton-root mantine-Button-root mantine-ulxcsg"
                data-button="true"
                data-loading="true"
                type="submit"
              >
                <div
                  class="mantine-1wpc1xj mantine-Button-inner"
                >
                  <span
                    class="mantine-Button-icon mantine-Button-leftIcon mantine-1jbrfp"
                  >
                    <svg
                      class="mantine-1avyp1d"
                      fill="#fff"
                      role="presentation"
                      style="width: calc(2.25rem / 2);"
                      viewBox="0 0 135 140"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <rect
                        height="120"
                        rx="6"
                        width="15"
                        y="10"
                      >
                        <animate
                          attributeName="height"
                          begin="0.5s"
                          calcMode="linear"
                          dur="1s"
                          repeatCount="indefinite"
                          values="120;110;100;90;80;70;60;50;40;140;120"
                        />
                        <animate
                          attributeName="y"
                          begin="0.5s"
                          calcMode="linear"
                          dur="1s"
                          repeatCount="indefinite"
                          values="10;15;20;25;30;35;40;45;50;0;10"
                        />
                      </rect>
                      <rect
                        height="120"
                        rx="6"
                        width="15"
                        x="30"
                        y="10"
                      >
                        <animate
                          attributeName="height"
                          begin="0.25s"
                          calcMode="linear"
                          dur="1s"
                          repeatCount="indefinite"
                          values="120;110;100;90;80;70;60;50;40;140;120"
                        />
                        <animate
                          attributeName="y"
                          begin="0.25s"
                          calcMode="linear"
                          dur="1s"
                          repeatCount="indefinite"
                          values="10;15;20;25;30;35;40;45;50;0;10"
                        />
                      </rect>
                      <rect
                        height="140"
                        rx="6"
                        width="15"
                        x="60"
                      >
                        <animate
                          attributeName="height"
                          begin="0s"
                          calcMode="linear"
                          dur="1s"
                          repeatCount="indefinite"
                          values="120;110;100;90;80;70;60;50;40;140;120"
                        />
                        <animate
                          attributeName="y"
                          begin="0s"
                          calcMode="linear"
                          dur="1s"
                          repeatCount="indefinite"
                          values="10;15;20;25;30;35;40;45;50;0;10"
                        />
                      </rect>
                      <rect
                        height="120"
                        rx="6"
                        width="15"
                        x="90"
                        y="10"
                      >
                        <animate
                          attributeName="height"
                          begin="0.25s"
                          calcMode="linear"
                          dur="1s"
                          repeatCount="indefinite"
                          values="120;110;100;90;80;70;60;50;40;140;120"
                        />
                        <animate
                          attributeName="y"
                          begin="0.25s"
                          calcMode="linear"
                          dur="1s"
                          repeatCount="indefinite"
                          values="10;15;20;25;30;35;40;45;50;0;10"
                        />
                      </rect>
                      <rect
                        height="120"
                        rx="6"
                        width="15"
                        x="120"
                        y="10"
                      >
                        <animate
                          attributeName="height"
                          begin="0.5s"
                          calcMode="linear"
                          dur="1s"
                          repeatCount="indefinite"
                          values="120;110;100;90;80;70;60;50;40;140;120"
                        />
                        <animate
                          attributeName="y"
                          begin="0.5s"
                          calcMode="linear"
                          dur="1s"
                          repeatCount="indefinite"
                          values="10;15;20;25;30;35;40;45;50;0;10"
                        />
                      </rect>
                    </svg>
                  </span>
                  <span
                    class="mantine-1ryt1ht mantine-Button-label"
                  >
                    Sign in
                  </span>
                </div>
              </button>
              <div
                class="mantine-Text-root mantine-wmjc1v"
              >
                Don't have an account? 
                <a
                  class="mantine-Text-root mantine-Anchor-root mantine-y73jqu"
                  href="/signup"
                >
                  Create an account
                </a>
              </div>
            </div>
          </form>
        </div>
      </main>
    </div>
  </div>
</DocumentFragment>
`;
